File: app/src/main/java/edu/application/ui/activities/MainActivity.java
package edu.application.ui.activities;

import androidx.appcompat.app.AppCompatActivity;
import androidx.navigation.NavController;
import androidx.navigation.fragment.NavHostFragment;
import androidx.navigation.ui.NavigationUI;

import android.annotation.SuppressLint;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;

import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

import edu.application.R;
import edu.application.databinding.ActivityMainBinding;
import edu.application.ui.adapters.TextFormatter;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;
    private NavHostFragment navHostFragment;
    private NavController navController;
    private BottomNavigationView bottomNavView;
    private SharedPreferences sharedPreferences;
    private FirebaseAuth auth;

    @SuppressLint("NonConstantResourceId")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        navHostFragment = (NavHostFragment)
                getSupportFragmentManager().findFragmentById(R.id.nav_host_fragment);
        assert navHostFragment != null;
        navController = navHostFragment.getNavController();
        navController.addOnDestinationChangedListener((navController, navDestination, bundle) -> {
            switch (navDestination.getId()) {
                case R.id.trainingFragment:
                case R.id.settingsFragment:
                case R.id.colorSchemeFragment:
                case R.id.colorSchemeCustomFragment:
                case R.id.fontSettingsFragment:
                case R.id.accountFragment:
                    showBottomNav();
                    break;

                default:
                    hideBottomNav();
                    break;
            }
        });

        bottomNavView = binding.bottomNavigation;
        NavigationUI.setupWithNavController(bottomNavView, navController);

        sharedPreferences = getPreferences(MODE_PRIVATE);
        TextFormatter.setSharedPreferences(sharedPreferences);

        auth = FirebaseAuth.getInstance();

        //if (sharedPreferences.getInt("auth", 0) == 0)
          //  navController.navigate(R.id.action_trainingFragment_to_loginFragment);
    }

    @Override
    protected void onStart() {
        super.onStart();

        FirebaseUser currentUser = auth.getCurrentUser();
        if(currentUser == null)
            navController.navigate(R.id.action_trainingFragment_to_loginFragment);
    }

    private void showBottomNav() {
        binding.bottomNavigation.setVisibility(View.VISIBLE);
    }

    private void hideBottomNav() {
        binding.bottomNavigation.setVisibility(View.GONE);
    }
}

File: app/src/main/kotlin/edu/application/ui/activities/MainActivity.kt
package edu.application.ui.activities

import android.content.SharedPreferences
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.NavController
import androidx.navigation.NavDestination
import androidx.navigation.fragment.NavHostFragment
import androidx.navigation.ui.NavigationUI.setupWithNavController
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.google.firebase.auth.FirebaseAuth
import edu.application.R
import edu.application.databinding.ActivityMainBinding
import edu.application.ui.fragments.navigation.adapters.TextFormatter

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private lateinit var navHostFragment: NavHostFragment
    private lateinit var navController: NavController
    private lateinit var bottomNavView: BottomNavigationView
    private lateinit var sharedPreferences: SharedPreferences
    private lateinit var auth: FirebaseAuth

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        navHostFragment =
            supportFragmentManager.findFragmentById(R.id.nav_host_fragment) as NavHostFragment
        navController = navHostFragment.navController
        navController.addOnDestinationChangedListener { _: NavController?,
                                                        navDestination: NavDestination,
                                                        _: Bundle? ->
            when (navDestination.id) {
                R.id.trainingFragment,
                R.id.settingsFragment,
                R.id.colorSchemeFragment,
                R.id.colorSchemeCustomFragment,
                R.id.fontSettingsFragment,
                R.id.accountFragment ->
                    showBottomNav()

                else -> hideBottomNav()
            }
        }

        bottomNavView = binding.bottomNavigation
        setupWithNavController(bottomNavView, navController)
        sharedPreferences = getPreferences(MODE_PRIVATE)
        TextFormatter.sharedPreferences = sharedPreferences

        auth = FirebaseAuth.getInstance()
    }

    override fun onStart() {
        super.onStart()

        val currentUser = auth.currentUser
        if (currentUser == null)
            navController.navigate(R.id.loginFragment)
    }

    private fun showBottomNav() {
        binding.bottomNavigation.visibility = View.VISIBLE
    }

    private fun hideBottomNav() {
        binding.bottomNavigation.visibility = View.GONE
    }
}